simpleflow/process/supervisor.py

```diff
@@ -106,10 +126,38 @@ def _handle_graceful_shutdown(signum, frame):
                 signal_name, os.getpid()))
             self.terminate()
 
+        # NB: Function is nested to have a reference to *self*.
+        def _handle_sigchld(signum, frame):
+            """
+            Handles SIGCHLD signal at supervisor process level. By construction this
+            handler is only attached to this process, not worker processes nor deciders.
+            """
+            logger.debug("process: caught signal signal=SIGCHLD, will cleanup zombies "
+                         "from pid={}".format(os.getpid()))
+
+            # cleanup children
+            # NB: here we use psutil's Process() API because it helps us identifying children
```

Comment:
I need another coffee: ppid is of course self.pid since we're alive... I tried testing `children()`, it didn't return zombies until it did. Paint me confused.